File: Java/src/ImplementStrStrTest.java
import org.junit.Assert;
import org.junit.Test;

public class ImplementStrStrTest {

    @Test
    public void testStrStr() {
        ImplementStrStr iss = new ImplementStrStr();

        Assert.assertTrue(iss.strStr("aaab", "b") == 3);

        Assert.assertTrue(iss.strStr("", "") == 0);

        Assert.assertTrue(iss.strStr("", "a") == -1);

        Assert.assertTrue(iss.strStr("whats up", "s ") == 4);
    }

}

File: Kotlin/src/ImplementStrStrTest.kt
import org.junit.Test

class ImplementStrStrTest {

    @Test
    fun testStrStr() {
        val iss = ImplementStrStr()

        assert(iss.strStr("aaab", "b") == 3)

        assert(iss.strStr("", "") == 0)

        assert(iss.strStr("", "a") == -1)

        assert(iss.strStr("whats up", "s ") == 4)
    }

}